### DISCUSSION TITLE
Update Component Title

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1486] [Proposals]
### Goal

Discutir e avaliar a adição de um título secundário (16px) nas opções existentes em Title.

### Context

Temos em nosso guia de recomendação o uso do text-preset-3 para títulos de módulos menores, como: como carrosséis de conteúdo, listas e agrupamentos de cards. 

Gostaríamos de acrescentar o mesmo como uma opção para títulos secundários.

**No Mística temos:**

Title 1 = 12px UPPERCASE

Title 2 = 18px

Title 3 = 24px

**Com isso:** 

Título de bloco = Title 1

Título primário = Title 2

Título primário = Title 3

Título secundário = Não existe correspondente no Mística

### Impact

Padronizar e disponibilizar o título que está sendo aplicado como secundário em módulos menores como um componente de título.

### Discussion Questions

ND

### Additional considerations

[Teste Playroom](https://mistica-web.vercel.app/playroom/#?code=N4Igxg9gJgpiBcIA8AFATjAbgSxgdwBUIIAbAZwAIALbWAPiQHp0tdDjy6AdAOyQCEIADwoAHAIZQo2HgHMAvMACMANgC%2B3HhQpIAygBdxYANYUyEsDEUAODbyQFs%2BkjCV1H%2BgK4kIFWBQAjH0gyJg8XN3twmAAmdydvX1E0bABbcRSIMKcXOKicmABmeK8fMRT0zOznIs0HGCF9M2wALysuEFUOinEyeQ7omI6SxLMYME8eKAzsLMYCBv1NbW0mAyNjOsZBIU1eXhAAGhB9KhhUmDIEAG0QADVsTAgAfQA5fBAAXWO8WlOr%2BDXQoqAAMnzUQA)

### Do you have a visual proposal?

https://www.figma.com/file/JkA7OefLqga6XyfiqLq1dY/Vivo---Lib-Intermedi%C3%A1ria?type=design&node-id=17109%3A32538&mode=design&t=DlRXybt1HyKdyAZM-1

**Comment by BrunoBiagioniNeto** [comment]
@yceballost ?

**Reply by yceballost** [reply]
Today we was talking about this discussion

The first thought was that there are fewww differences between Title2 and this new "title4 / titleSecondary or whatever" with 16px
![image](https://github.com/Telefonica/mistica-design/assets/6722153/2fb9f7d8-bc4e-4488-acc7-3ce708200ffe)

We may run into the problem of having many similar sizes and have hierarchy problems.

The hierarchy of titles is more difficult than it may seem 😂 (more when some skins have different sizes from the rest of the skins

We talked about the possibility of creating a new `<Title>` component in the code with the flexibility to use any size inside and keeping the slot in the right side to cover these kinds of needs, we think it is the right way at the moment.

@aweell anything to include?

**Reply by annefranca** [reply]
@yceballost então teríamos apenas um componente Title que poderia receber qualquer valor?

**Reply by aweell** [reply]
Hey @annefranca 

The rest of the titles can still use the current ones (title1 / 2 / 3). For the title secondary or any other titles that may appear the solution of having a title that receives the size and semantic hierarchy would solve the problem. In the playroom example of the request you were using a text primitive component to simulate how the title would look like, the idea is similar but allowing also to have the right prop so like other titles, you can provide a link to "see more" or any other purpose.

**Reply by annefranca** [reply]
@aweell essa seria uma evolução no Mística? Você tem algum link ou arquivo com essa proposta?

**Reply by annefranca** [reply]
@yceballost @aweell temos alguma atualização?

**Reply by annefranca** [reply]
@yceballost @aweell temos alguma atualização? Como ficaria a aplicação no Figma e a aplicação no código?

**Comment by aweell** [comment]
Hi @annefranca,

After evaluating internally the possibility of having a more agnostic `title` component we decided that we won't add that in the system. Opening the possibility of having personalization in the title sizes or creating new elements could lead to a lack of consistency and relationships when building global products.

As @yceballost commented previously the differences in hierarchy found between `title2` and the new title proposal are minimal and we believe that the current scale of titles available + other elements that help to create hierarchy are enough to cover most product cases sucessfully.

The solution we propose for this particular case is the usage of the `text` component as a title using the semantic transformation property that is currently available (`as`) following the playroom test that the OP included.

Example -> https://tinyurl.com/ywuca3ec
